### ISSUE TITLE
Use pickle to unknown objects

### ISSUE THREAD

**Issue by turicas** [turicas/mongodict#5]
Currently it'll work only for objects `pymongo` can store (as key or as value). It should pickle other types of objects.

**Comment by turicas** [comment]
I'm considering in do not implementing this issue. I think it should be in a level on top of `mongodict` (let the developer do this work outside `mongodict`). Summarizing possible problems:
- If `mongodict` tries automatically to pickle the object, it can lead to errors like outdated objects saved in the database (see the example below).
- We'd need a kind of "protocol" to discover which objects are pickled and which are not. It can be a mess (imagine the collection with lots of non-homogeneous data, but with the same schema).

The problems involving pickled objects are exemplified below:

``` python
from mongodict import MongoDict


class MyClass(object):
    pass

obj = MyClass()
obj.test = 123
my_dict = MongoDict()
my_dict['obj'] = obj
obj.python = 'rules'
assert my_dict['obj'] == obj # False
```

(I think) ZODB solves this problem forcing the developer to subclass a specific `Persistent` class, so it'll update object info in the database when you update the object automatically but as `mongodict` **is not** a object-oriented database, we can't force the developer to do it (maybe a library on top of `mongodict` can do it).

It's open for discussion.

**Comment by turicas** [comment]
Implemented on 194333fbe88a400243a510fd9a5b23f1f2bc62fc.
